lib/sqlite.js

```diff
@@ -1,5 +1,15 @@
 'use strict';
+const {
+  globalThis,
+} = primordials;
+
 const { emitExperimentalWarning } = require('internal/util');
 
 emitExperimentalWarning('SQLite');
 module.exports = internalBinding('sqlite');
+
+const { Iterator } = globalThis;
```

Comment:
I would prefer if we could figure out a way to do this without introducing JS code into this API.

Reply:
I explored `SafeArrayIterator` which is not working.
In C++ code, I tries some things with v8 API so the iterator extends `Iterator` without sucess.

```cpp
  auto JSIteratorPrototype = context->Global()->Get(context, String::NewFromUtf8Literal(isolate, "Iterator.prototype")).ToLocalChecked();
  iterableIterator->SetPrototype(context, JSIteratorPrototype).Check();
```

I'm not experienced with cpp, I discover a bit of v8 for this PR.
If someone can purpose me solution, in cpp, so `StatementSync::Iterate` return a complete JS iterator (extend Iterator, integrated with IteratorHelpers). I will be very happy to do it.

Reply:
What you did probably translates to `globalThis['Iterator.prototype']` in JS, so I'm not surprised it doesn't work. You probably need to do it in two steps.

Reply:
I suppose it could be better:

```cpp
  Local<Object> global = context->Global();
  Local<Object> globalThis = global->Get(context,  String::NewFromUtf8Literal(isolate, "globalThis")).ToLocalChecked().As<Object>();
  Local<Object> JSIterator = globalThis->Get(context,  String::NewFromUtf8Literal(isolate, "Iterator")).ToLocalChecked().As<Object>();
  Local<Object> JSIteratorPrototype = JSIterator->Get(context,  String::NewFromUtf8Literal(isolate, "prototype")).ToLocalChecked().As<Object>();

  iterableIterator->SetPrototype(context, JSIteratorPrototype).ToChecked();
```

But it works